element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpCorrelationTest.java

```diff
@@ -0,0 +1,322 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation. 
+ ******************************************************************************/
+package org.eclipse.californium.elements.tcp;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.text.IsEmptyString.isEmptyOrNullString;
+import static org.junit.Assert.assertArrayEquals;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.californium.elements.Connector;
+import org.eclipse.californium.elements.CorrelationContext;
+import org.eclipse.californium.elements.RawData;
+import org.eclipse.californium.elements.TcpCorrelationContext;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class TcpCorrelationTest {
+
+	@Rule
+	public final Timeout timeout = new Timeout(20, TimeUnit.SECONDS);
+
+	private final List<Connector> cleanup = new ArrayList<>();
+
+	@After
+	public void cleanup() {
+		for (Connector connector : cleanup) {
+			connector.stop();
+		}
+	}
+
+	@Test
+	public void correlationContext() throws Exception {
```

Comment:
Can you please add a JavaDoc comment indicating what this test case is supposed to verify?
This makes it much easier to a) understand and b) review the code :-)

Same holds true for the other test cases as well...